app/src/main/kotlin/org/kotlinlsp/index/worker/WorkerThread.kt

```diff
@@ -26,36 +26,49 @@ class WorkerThread(
         while(true) {
             when(val command = workQueue.take()) {
                 is Command.Stop -> break
+                is Command.ScanFile -> {
+                    if(command.virtualFile.url.startsWith("file://")) {
+                        val ktFile = project.read { PsiManager.getInstance(project).findFile(command.virtualFile) } as KtFile
+                        scanKtFile(project, ktFile, db)
+                    } else {
+                        // TODO
```

Comment:
We are going to scan only source files, so they always start with `file://`. Scanning libraries is not needed for the analysis api to work (the analysis api does this for us as we have configured in `PlatformSettings.deserializedDeclarationsOrigin = KotlinDeserializedDeclarationsOrigin.BINARIES`, so it reads .jar files when needed)

Reply:
Just for curiosity, we could configure it to be `PlatformSettings.deserializedDeclarationsOrigin = KotlinDeserializedDeclarationsOrigin.STUBS` (that's how it's configured in the kotlin ide plugin), in that case we would need to scan libraries as well, but we are not doing that for simplicity, we might do it in the future to improve performance and avoid repeated processing